packages/catalog-client/src/types/status.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  UNSTABLE_EntityStatusLevel,
+  UNSTABLE_EntityStatusValue,
+} from '@backstage/catalog-model';
+import { SerializedError } from '@backstage/errors';
+
+/*
+ * This is the entity status field that's emitted by the catalog processing
+ * engine, to inform about the status of an entity.
+ *
+ * Example:
+ *
+ * "status": {
+ *   "backstage.io/catalog-processing": {
+ *     "status": "error",
```

Comment:
What would happen if items include a mix of warnings and ok? Will status always be the most severe status?

Reply:
Up to the catalog implementation (this is not something generic for all sources), but yes, that's the intention!

Reply:
Feeling we should skip this just to keep the data more normalized. If we end up feeling that there's a lot of tedious checking for the same thing we could add it back later.

Reply:
Not though that this is the one normalized part :) if we don't have a root field like this, we either:

- have to cast to specific types in each consumer (eg the entities table wanting to show status dots), or
- force the `items` structure on all specific types, which I felt was unfortunate

Reply:
So to be clear, this field appears in the "base type" for all statuses on this pr

Reply:
Another angle could be to completely change how this works and make it an array rather than an object. Similar to k8s status conditions:

```yaml
status:
  conditions:
    - source: backstage.io/catalog-processing
      status: "error"
      error: { ... }
    - source: backstage.io/catalog-processing
      status: "warning"
      message: "The owner Group:default/missplelled does not exist in the catalog"
    - source: pagerduty.com/alert-status
      status: "warning"
      message: "..."
```

Reply:
To solidify the discord discussion:

```yaml
status:
  conditions:
    - type: backstage.io/catalog-processing
      level: error
      message: "..."
      ... perhaps additional fields in the future for tracing etc.
    - type: backstage.io/catalog-processing
      level: warning
      message: "The owner Group:default/missplelled does not exist in the catalog"
    - type: pagerduty.com/alert-status
      level: warning
      message: "..."
```